### PR TITLE
Add shell script to generate analysis files that are included in data releases

### PR DESCRIPTION
**Pull request by jaclyn-taroni**
⚠️  Review #1419 and #1420 first ⚠️  this can not be stacked because these are from my branch.

Continuing to break up the changes described in https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/1399#issuecomment-1124329628

Here I'm adding `scripts/generate-analysis-files-for-release.sh`, a shell script for generating **analysis files** that are included in the data download. As a reminder, [an analysis file is defined as](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/d2f2c6e1786dabb0dcf5aa099a203c2baef5c8b9/doc/data-files-description.md):

> Any file created by a script in `analyses/*`.

It is intended to be used as part of a series of steps during the data release process, which will be as follows:

1. We start a release that has all of the PBTA data files (i.e., upstream files) included
2. We run `scripts/generate-analysis-files-for-release.sh`, which should generate all the analysis files for a release and put them in `scratch/analysis_files_for_release`, and commit any changes to files that are included in the repository. 
3. We add all the analysis files to the release.
4. We run `scripts/run-for-subtyping.sh` and commit any changes to files that are included in the repository. (PR coming soon!)
5. We add `pbta-histologies.tsv` to the release

### CODE REVIEW

**Review by jharenza — APPROVED** [review]
👍

File: scripts/generate-analysis-files-for-release.sh [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  cp ${analyses_dir}/snv-callers/results/consensus/pbta-snv-mutation-tmb-coding.tsv ${compiled_dir}
+  cp ${analyses_dir}/snv-callers/results/consensus/pbta-snv-mutation-tmb-all.tsv ${compiled_dir}
+  ## TCGA
+  cp ${analyses_dir}/snv-callers/results/consensus/tcga-snv-consensus-snv.maf.tsv.gz ${compiled_dir}
+  cp ${analyses_dir}/snv-callers/results/consensus/tcga-snv-mutation-tmb-coding.tsv ${compiled_dir}
+  cp ${analyses_dir}/snv-callers/results/consensus/tcga-snv-mutation-tmb-all.tsv ${compiled_dir}
+
+  # Run hotspot detection
+  echo "Run hotspots detection"
+  bash ${analyses_dir}/hotspots-detection/run_overlaps_hotspots.sh
```
Comment (sjspielman) [inline scripts/generate-analysis-files-for-release.sh:81]: Noting that this module (and output file `pbta-snv-scavenged-hotspots.maf.tsv.gz`) appears to be missing from [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/doc/data-files-description.md) but it should be there, right?

File: scripts/generate-analysis-files-for-release.sh [hunk 2]
```diff
@@ -0,0 +1,123 @@
+#!/bin/sh
+
+set -e
+set -o pipefail
+
+# Set the working directory to the directory of this file
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+# If RUN_LOCAL is used, the time-intensive steps are skipped because they cannot
```
Comment (sjspielman) [inline scripts/generate-analysis-files-for-release.sh:9]: and _memory_-intensive!

File: scripts/generate-analysis-files-for-release.sh [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+OPENPBTA_BASE_RELEASE=1 bash ${analyses_dir}/fusion_filtering/run_fusion_merged.sh
+
+# Fusion summary
+echo "Run fusion summary for subtypes"
+bash ${analyses_dir}/fusion-summary/run-new-analysis.sh
+
+# Compile all the files that need to be included in the release in one place
+# in the scratch directory
+compiled_dir=${scratch_dir}/analysis_files_for_release
+mkdir -p ${compiled_dir}
```
Comment (sjspielman) [inline scripts/generate-analysis-files-for-release.sh:42]: I would move these lines up before you start running modules, right after other directories are defined in lines 19-21

File: scripts/generate-analysis-files-for-release.sh [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  # Run consensus CN caller step
+  echo "Run CNV consensus"
+  bash ${analyses_dir}/copy_number_consensus_call/run_consensus_call.sh
+
+  # Copy over CNV consensus
+  cp ${analyses_dir}/copy_number_consensus_call/results/pbta-cnv-consensus.seg.gz ${compiled_dir}
+
+  # Run GISTIC step -- only the part that generates ZIP file
+  echo "Run GISTIC"
+  # Run a step that subs ploidy for NA to allow GISTIC to run
```
Comment (sjspielman) [inline scripts/generate-analysis-files-for-release.sh:95]: Just a question about the whether the CI if lines might be needed here.

https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/e31b58ce6d6f01409a2fdef0191d6f8b6126a668/analyses/run-gistic/run-gistic-module.sh#L9-L32

File: scripts/generate-analysis-files-for-release.sh (hunk 1 above)
Comment (jaclyn-taroni) [reply scripts/generate-analysis-files-for-release.sh:81]: Yes they should, but maybe that should get fixed on #1424 and not here?

Comment (sjspielman) [reply scripts/generate-analysis-files-for-release.sh:81]: I'll open an issue

File: scripts/generate-analysis-files-for-release.sh (hunk 4 above)
Comment (jaclyn-taroni) [reply scripts/generate-analysis-files-for-release.sh:95]: I don't follow. I don't necessarily expect this script (`scripts/generate-analysis-files-for-release.sh`) to get run in CI. 

We don't use `analyses/run-gistic/run-gistic-module.sh` at all here.

Comment (sjspielman) [reply scripts/generate-analysis-files-for-release.sh:95]: > I don't necessarily expect this script (scripts/generate-analysis-files-for-release.sh) to get run in CI.

Just looking for a confirmation of this! 👍

**Review by sjspielman — APPROVED** [review]
The remaining small feedback items I left are quick and at your discretion, so I don't need to see again 🚀

**Pull request merged**